### PR TITLE
Remove DefaultEvaluator shap plot de-duplicating feature names logic and fix color bar

### PR DESCRIPTION
**Pull request by WeichenXu123**
Signed-off-by: Weichen Xu <weichen.xu@databricks.com>

## Related Issues/PRs

#xxx

## What changes are proposed in this pull request?

Remove DefaultEvaluator shap plot de-duplicating feature names logic,
According to suggestion https://github.com/mlflow/mlflow/pull/6705#issuecomment-1237829665
and fix color bar.

Closes #6705

## How is this patch tested?

manually.
shap_beeswarm_plot_on_data_diabetes_dataset
shap_feature_importance_plot_on_data_diabetes_dataset
shap_summary_plot_on_data_diabetes_dataset

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Click the `Details` link on the `Preview docs` check.
2. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

Remove DefaultEvaluator shap plot de-duplicating feature names logic

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [x] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Comment by harupy** [comment]
@WeichenXu123 Can you paste the plot images in the PR description?

**Comment by harupy** [comment]
Not sure if you notice, but we need to run the following code before saving the image:

```
plt.xticks(fontsize=10)
plt.yticks(fontsize=10)
plt.savefig("beeswarm.png", bbox_inches="tight")
```

**Comment by harupy** [comment]
We can remove tests for feature name truncation.

File: tests/models/test_default_evaluator.py [hunk 1]
```diff
@@ -658,6 +658,7 @@ def _evaluate_explainer_with_exceptions(model_uri, dataset):
                 "ignore_exceptions": False,
             },
         )
+    print(f"runid={run.info.run_id}\n")
```
Comment (harupy) [inline tests/models/test_default_evaluator.py:661]:
```suggestion
```

File: mlflow/models/evaluation/default_evaluator.py [hunk 2]
```diff
@@ -474,7 +473,9 @@ def _log_image_artifact(
         try:
             pyplot.clf()
             do_plot()
-            pyplot.savefig(artifact_file_local_path)
+            pyplot.xticks(fontsize=10)
+            pyplot.yticks(fontsize=10)
+            pyplot.savefig(artifact_file_local_path, bbox_inches="tight")
```
Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:478]: We should not call this here.

Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py:478]: You can call this right after creating SHAP plots.

File: tests/models/test_default_evaluator.py [hunk 3]
```diff
@@ -1516,29 +1517,6 @@ def test_autologging_is_disabled_during_evaluate(model):
         mlflow.sklearn.autolog(disable=True)
 
 
-def test_truncation_works_for_long_feature_names(linear_regressor_model_uri, diabetes_dataset):
```
Comment (harupy) [inline tests/models/test_default_evaluator.py:1519]: You can also remove `iris_pandas_df_longname_cols_dataset`.

File: tests/models/test_default_evaluator.py [hunk 4]
```diff
@@ -646,7 +646,7 @@ def test_svm_classifier_evaluation(svm_model_uri, breast_cancer_dataset, baselin
 
 
 def _evaluate_explainer_with_exceptions(model_uri, dataset):
-    with mlflow.start_run():
+    with mlflow.start_run() as run:
```
Comment (harupy) [inline tests/models/test_default_evaluator.py:649]:
```suggestion
    with mlflow.start_run():
```

File: mlflow/models/evaluation/default_evaluator.py [hunk 5]
```diff
@@ -669,9 +659,14 @@ def _log_model_explainability(self):
             )
             _logger.debug("", exc_info=True)
 
+        def _adjust_color_bar():
+            pyplot.gcf().axes[-1].set_aspect('auto')
+            pyplot.gcf().axes[-1].set_box_aspect(50)
+
         def plot_beeswarm():
             pyplot.subplots_adjust(bottom=0.2, left=0.4)
```
Comment (harupy) [inline mlflow/models/evaluation/default_evaluator.py:667]: Do we need this line?

File: mlflow/models/evaluation/default_evaluator.py [hunk 6]
```diff
@@ -669,9 +659,14 @@ def _log_model_explainability(self):
             )
             _logger.debug("", exc_info=True)
 
+        def _adjust_color_bar():
+            pyplot.gcf().axes[-1].set_aspect('auto')
+            pyplot.gcf().axes[-1].set_box_aspect(50)
```
Comment (WeichenXu123) [inline mlflow/models/evaluation/default_evaluator.py:664]: This fixed color bar renderring:

Before:
shap_summary_plot_on_data_iris_pandas_df_longname_cols_dataset

After:
shap_summary_plot_on_data_iris_pandas_df_longname_cols_dataset

File: mlflow/models/evaluation/default_evaluator.py (hunk 5 above)
Comment (WeichenXu123) [reply mlflow/models/evaluation/default_evaluator.py]: No. Now it is auto size layout . let me remove it.

File: mlflow/models/evaluation/default_evaluator.py (hunk 2 above)
Comment (WeichenXu123) [reply mlflow/models/evaluation/default_evaluator.py:479]: like this ?
```
            pyplot.clf()
            pyplot.xticks(fontsize=10)
            pyplot.yticks(fontsize=10)
            do_plot()
            pyplot.savefig(artifact_file_local_path, bbox_inches="tight")
```

Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py]: No.

Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py]: I meant in `plot_beeswarm` and `plot_beeswarm`.

Comment (WeichenXu123) [reply mlflow/models/evaluation/default_evaluator.py]: emm, after apply change https://github.com/mlflow/mlflow/pull/6721#discussion_r964479412 the plot becomes
shap_beeswarm_plot_on_data_diabetes_dataset

Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py:479]: I think this is because the image size is fixed.

Comment (WeichenXu123) [reply mlflow/models/evaluation/default_evaluator.py:479]: updated according to your suggestion.

Comment (harupy) [reply mlflow/models/evaluation/default_evaluator.py]: I meant you should do:

```
        def plot_beeswarm():
            shap.plots.beeswarm(shap_values, show=False, color_bar=True)
            pyplot.xticks(fontsize=10)
            pyplot.yticks(fontsize=10)
            pyplot.savefig(artifact_file_local_path, bbox_inches="tight")
            _adjust_color_bar()
```

**Comment by WeichenXu123** [comment]
> Did you find a way to fix [#6721 (comment)](https://github.com/mlflow/mlflow/pull/6721#discussion_r964486618)?

No issue, current code works fine. It is my previous code issue.

File: mlflow/models/evaluation/default_evaluator.py (hunk 2 above)
Comment (WeichenXu123) [reply mlflow/models/evaluation/default_evaluator.py]: yes. that's what current code does.

**Review by harupy — APPROVED** [review]
LGTM

**Pull request merged**